File: aoc2020/day1/part2.py
def main():
    with open("input.txt", "r") as f:
        p = f.read().split("\n")

    p = [int(i) for i in p]
    res = 1
    for item1 in p:
        for item2 in p[::-1]:
            for item3 in p:
                if item1 + item2 + item3 == 2020:
                    res = item1 * item2 * item3
                    break

    print(res)


if __name__ == "__main__":
    main()

File: aoc2020/src/day1/part2.rs
use std::env;
use std::fs;

fn main() {
    let contents = fs::read_to_string("./input.txt").expect("Failed to read file");
    let contents = contents.split("\n");
    let list: Vec<i64> = contents
        .map(|s| s.parse::<i64>())
        .filter_map(Result::ok)
        .collect();

    for i in 0..list.len() {
        for j in 0..list.len() {
            for k in 0..list.len() {
                if list[i] + list[j] + list[k] == 2020 {
                    println!("{}", list[i] * list[j] * list[k]);
                    break;
                }
            }
        }
    }
}
